models/repo.go

```diff
@@ -1891,21 +1891,20 @@ func DeleteRepository(doer *User, uid, repoID int64) error {
 	return nil
 }
 
-// GetRepositoryByRef returns a Repository specified by a GFM reference.
-// See https://help.github.com/articles/writing-on-github#references for more information on the syntax.
-func GetRepositoryByRef(ref string) (*Repository, error) {
-	n := strings.IndexByte(ref, byte('/'))
-	if n < 2 {
-		return nil, ErrInvalidReference
-	}
-
-	userName, repoName := ref[:n], ref[n+1:]
-	user, err := GetUserByName(userName)
+// GetRepositoryByOwnerAndName returns the repository by given ownername and reponame.
+func GetRepositoryByOwnerAndName(ownerName, repoName string) (*Repository, error) {
+	var repo Repository
+	has, err := x.Select("repository.*").
+		Join("INNER", "user", "`user`.id = repository.owner_id").
+		Where("repository.lower_name = ?", strings.ToLower(repoName)).
+		And("`user`.lower_name = ?", strings.ToLower(ownerName)).
+		Get(&repo)
 	if err != nil {
 		return nil, err
+	} else if !has {
+		return nil, ErrRepoNotExist{0, 0, ownerName + "/" + repoName}
```

Comment:
nit: Elsewhere in the code, the `Name` field of the `ErrRepoNotExist` struct is just the repo name (e.g. in `GetRepositoryByName`), but here it is both the owner's name and repo name. Could we instead add a `OwnerName` field, and keep the `Name` field only for the repo name?